lib/developer_portal/app/controllers/developer_portal/admin/account/payment_details_base_controller.rb

```diff
@@ -27,7 +27,12 @@ def edit
 
   def update
     current_account.updating_payment_detail = true
-    if current_account.update account_params
+    billing_details_updated = current_account.transaction do
+      raise ActiveRecord::Rollback unless current_account.update(account_params) && update_address_on_payment_gateway
+
+      true
+    end
```

Comment:
Is the point here to update both or none? Is it not valid to update only one of those?

Reply:
Yeah, that's the idea. It comes out of this discussion: https://github.com/3scale/porta/pull/3696#discussion_r1634386215

So I added it explicitly: https://github.com/3scale/porta/pull/3696/commits/c049e437b82405cbd948cea8343f7df685bf8699

Certainly, Stripe should not change if our local model has not been updated.

Whether it's OK to update our model, but not Stripe - this can be argued. Previously updating billing address in the model was not reflected in Stripe in any way.
Also, it is not clear what should we tell the user if the address has been updated on the model (in our DB), but not in Stripe - is it a success or not?

So, in the current implementation if any of the two updates fail, the data is not saved and the user is notified about the error.